src/Integration.Vsix/SonarLintTagger/AccumulatingIssueConsumer.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using SonarLint.VisualStudio.Core.Analysis;
+
+/* 
+ * Instancing: a new instance of this class should be created for each analysis request.
+ * 
+ * The class is initialized with the text snapshot representing the state of the text buffer
+ * at the point the analysis was triggered ("the analysis snaphost").
+ * 
+ * Each time IIssueConsumer.Accept is called, the new issues will be mapped back to the
```

Comment:
Maybe "mapped back to the _given_ analysis snapshot"?

Reply:
To me that's more confusing - what "the analysis snapshot" is is defined in line 32. Saying the "given analysis snapshot" suggests there could be more than one "analysis snapshot".

I'll drop the reference to "analysis snapshot" and change it to "supplied snapshot"

Reply:
"supplied" sounds good.